src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -145,6 +132,19 @@ public void RequestAnalysis(IAnalyzerOptions options)
             }
         }
 
+        private void RequestNewAnalysis(IAnalyzerOptions options)
+        {
+            LastAnalysisFilePath = document.FilePath; // Refresh the stored file path in case the document has been renamed
+            var analysisSnapshot = document.TextBuffer.CurrentSnapshot;
+            NotifyFileTracker(analysisSnapshot);
+            vsAwareAnalysisService.RequestAnalysis(document, LastAnalysisFilePath, analysisSnapshot, document.Encoding, detectedLanguages, SnapToNewSnapshot, options);
```

Comment:
The intention seem to be to capture the snapshot of the analysis and not use the information directly from the ITextDocument. 

Please:

1. create a record/class called AnalysisSnapshot that has the properties: FilePath, Snapshot, Encoding. 
2. add a test that we used the captured data

There are two reasons for this:

- We make the intention explicit
- We prevent increasing the number of parameters that we pass around in different classes (which are also confusing, because all that data can be taken from the ITextDocument)
- We can test